rocketpool-cli/service/config/addon-apcupsd.go

```diff
@@ -0,0 +1,174 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+	"github.com/rocket-pool/smartnode/shared/services/config"
+	"github.com/rocket-pool/smartnode/shared/types/addons"
+)
+
+// The page wrapper for the APCUPSD addon config
+type AddonApcupsdPage struct {
+	addonsPage     *AddonsPage
+	page           *page
+	layout         *standardLayout
+	masterConfig   *config.RocketPoolConfig
+	addon          addons.SmartnodeAddon
+	enabledBox     *parameterizedFormItem
+	modeBox        *parameterizedFormItem
+	exporterImage  *parameterizedFormItem
+	apcupsdImage   *parameterizedFormItem
+	mountPoint     *parameterizedFormItem
+	metricsPort    *parameterizedFormItem
+	apcupsdAddress *parameterizedFormItem
+}
+
+// Creates a new page for the APCUPSD addon settings
+func NewAddonApcupsdPage(addonsPage *AddonsPage, addon addons.SmartnodeAddon) *AddonApcupsdPage {
+
+	configPage := &AddonApcupsdPage{
+		addonsPage:   addonsPage,
+		masterConfig: addonsPage.home.md.Config,
+		addon:        addon,
+	}
+	configPage.createContent()
+
+	configPage.page = newPage(
+		addonsPage.page,
+		"settings-addon-apcupsd",
+		addon.GetName(),
+		addon.GetDescription(),
+		configPage.layout.grid,
+	)
+
+	return configPage
+
+}
+
+// Get the underlying page
+func (configPage *AddonApcupsdPage) getPage() *page {
+	return configPage.page
+}
+
+// Creates the content for the APCUPSD settings page
+func (configPage *AddonApcupsdPage) createContent() {
+
+	// Create the layout
+	configPage.layout = newStandardLayout()
+	configPage.layout.createForm(&configPage.masterConfig.Smartnode.Network, fmt.Sprintf("%s Settings", configPage.addon.GetName()))
+
+	// Return to the home page after pressing Escape
+	configPage.layout.form.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		if event.Key() == tcell.KeyEsc {
+			// Close all dropdowns and break if one was open
+			for _, param := range configPage.layout.parameters {
+				dropDown, ok := param.item.(*DropDown)
+				if ok && dropDown.open {
+					dropDown.CloseList(configPage.addonsPage.home.md.app)
+					return nil
+				}
+			}
+
+			// Return to the home page
+			configPage.addonsPage.home.md.setPage(configPage.addonsPage.page)
+			return nil
+		}
+		return event
+	})
+
+	// Get the parameters
+	enabledParam := configPage.addon.GetEnabledParameter()
+	// TODO: Don't like how I reference these by index here. Is there a better way?
```

Comment:
Any suggestion for a cleaner way to fetch these parameters? It seems I can only access functions as defined by the generic addon interface.